datalad/distribution/create_publication_target_sshwebserver.py

```diff
@@ -0,0 +1,270 @@
+        lgr.debug("Try starting control master by calling:\n%s" % cmd)
+        import subprocess
+        proc = subprocess.Popen(cmd, shell=True)
+        proc.communicate(input="\n")  # why the f.. this is necessary?
+
+        runner = Runner()
+        ssh_cmd = ["ssh", "-S", control_path, host_name]
+
+        for repo in repos:
+
+            if target in repos[repo].git_get_remotes():
+                # TODO: skip or raise? If raise, do it before looping, in order
+                # to fail without messing up things by partially doing the shit
+                # - what about just adding new push url to that remote?
+                # - if url(s) fit, then still try to create target?
+
+                # cmd = ["git", "remote", "get-url", "--push", remote]
+                # out, err = runner.run(cmd, cwd=handle_repo.path)
+                # if handle_remote_url_push != out.strip():
+                lgr.warning("Sibling {0} already exists. Skipping".format(target))
+                continue
+
+            # %NAME
+            REPO_NAME = repo.replace("/", "-")
+
+            # create remote repository
+            path = target_dir.replace("%NAME", REPO_NAME)
+
+            if path != '.':
+                # check if target exists, and if not --force is given,
+                # fail here
+                # TODO: Is this condition valid for != '.' only?
+                path_exists = True
+                cmd = ssh_cmd + ["ls", path]
+                try:
+                    out, err = runner.run(cmd, expect_fail=True,
+                                          expect_stderr=True)
+                except CommandError as e:
+                    if "%s: No such file or directory" % path in e.stderr:
+                        path_exists = False
+                    else:
+                        raise  # It's an unexpected failure here
+
+                if path_exists and not force:
+                    raise RuntimeError("Target directory %s already exists." %
+                                       path)
+
+                cmd = ssh_cmd + ["mkdir", "-p", path]
+                try:
+                    runner.run(cmd)
+                except CommandError as e:
+                    lgr.error("Remotely creating target directory failed at "
+                              "%s.\nError: %s" % (path, str(e)))
+                    continue
+
+            # init git repo
+            cmd = ssh_cmd + ["git", "-C", path, "init"]
+            try:
+                runner.run(cmd)
+            except CommandError as e:
+                lgr.error("Remotely initializing git repository failed at %s."
+                          "\nError: %s" % (path, str(e)))
+                continue
+
+            # allow for pushing to checked out branch
+            cmd = ssh_cmd + ["git", "-C", path, "config",
+                             "receive.denyCurrentBranch",
+                             "updateInstead"]
```

Comment:
That ping-pong would be a solution, but not within this command. Because it needs a create-target, publish, clean-target or sth. like that. So either we let create checkout some temporary branch in that case and have message, that tells the user to also call a post-publish command or have this mechanic in publish command. Publish could have an option to do this, when it cannot push to the target.